cylc/flow/simulation.py

```diff
@@ -166,46 +220,46 @@ def sim_time_check(
     Returns:
         True if _any_ simulated task state has changed.
     """
-    sim_task_state_changed = False
-    now = time()
+    sim_task_state_changed: bool = False
+    now: float = time()
     for itask in itasks:
         if itask.state.status != TASK_STATUS_RUNNING:
             continue
         # Started time is not set on restart
-        if itask.summary['started_time'] is None:
-            itask.summary['started_time'] = now
-        timeout = (
-            itask.summary['started_time'] +
-            itask.tdef.rtconfig['simulation']['simulated run length']
-        )
+        started_time = itask.summary['started_time']
+        if started_time is None:
+            itask.summary['started_time'] = started_time = now
+        if itask.mode_settings is None:
+            itask.mode_settings = ModeSettings()
+            itask.mode_settings.update(
+                itask, broadcast_mgr)
+        timeout = started_time + itask.mode_settings.simulated_run_length
```

Comment:
These values should get set the first time `sim_time_check` is called on a simulated job.

On each subsequent call the values are retrieved from the itask so should stay the same.

Worth considering setting these values during simulated job submission for clearer code.

Reply:
What happens if this is a restart?

Reply:
You tell me!